### ISSUE TITLE
WO-005: production

### ISSUE THREAD

**Issue by Xocas12** [Xocas12/gosplan-env#10]
> **WO-005** - Phase 1 - Owner **MID-strong** - Difficulty **3/5** - Gate **G0**
> Card: [`workorders/WO-005.md`](https://github.com/Xocas12/gosplan-env/blob/main/workorders/WO-005.md)

## What this is

This card implements the physical side of an enterprise: how much it actually makes in one PRODUCE
step, given the effort it chose, the inputs it has on hand and a random yield shock, and what that
effort costs it. Two bodies in `gosplan/env/production.py`: `coverage`, the CES aggregator that turns
per-good input availability into a single scarcity multiplier in `[0, 1]`, and `produce_step`, which
runs one step for all `N` enterprises and consumes inputs against output.

It exists so that scarcity has physical consequences rather than accounting ones. This module knows
nothing about targets, reports, penalties or rewards - it computes output and cost, and nothing else.

## Depends on

- WO-003
- WO-004

The frozen suite this card must pass is WO-002's.

**Blocked by:** #8 (WO-003), #9 (WO-004)
This card may not be started until every issue above is closed (CONTRACT rule 12 / the loop in CONTRIBUTING.md).

## Read first (exhaustive whitelist)

- [`spec/spec.py`](https://github.com/Xocas12/gosplan-env/blob/main/spec/spec.py) - frozen signatures
  and docstrings of `coverage` and `produce_step`.
- [`gosplan/env/production.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/production.py) -
  **the write target.** Module docstring, `Array` alias and the full contract of both functions are
  already there; two bodies missing.
- [`gosplan/config.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/config.py) -
  `EnvConfig` and the SUPPLY / INC fields this card reads: `productivity`, `io_matrix`, `sector_of`,
  `yield_sigma`, `input_complementarity`, `setup_cost`, `irs_alpha`, `delivery_timing`,
  `arrival_probs`, `quality_cost`, `quality_matters`, `steps_per_period`, `effort_cost`.
- [`gosplan/rng.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/rng.py) - `draw` and
  `PURPOSES`; the only source of randomness allowed here.
- [`gosplan/env/state.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/state.py) -
  `State` and `EnterpriseAction` field names, shapes and semantics.
- `docs/ref_worked_example.md` - the hand-validated two-enterprise, two-sector trajectory; the
  numbers this card must reproduce (written by WO-002).
- [`CONTRACT.md`](https://github.com/Xocas12/gosplan-env/blob/main/CONTRACT.md) - rules 3, 7 (no
  hard-coded pathology), 9 (RNG), 12.
- [`pyproject.toml`](https://github.com/Xocas12/gosplan-env/blob/main/pyproject.toml) - ruff
  line-length 100, target py312.
- [`tests/unit/test_production.py`](https://github.com/Xocas12/gosplan-env/blob/main/tests/unit/test_production.py) -
  read-only (CONTRACT rule 2); written by WO-002, so it is not in the tree until that card lands.

Read nothing else. CONTRACT rule 12.

## Write only

- `gosplan/env/production.py`

Creating, editing or deleting any other file is a CONTRACT rule 12 violation.

## Must pass

- [ ] `tests/unit/test_production.py` - **T-U7**: `theta = inf` equals `min`; `theta -> 1` equals the
      weighted harmonic mean; `H = 1` when no inputs are needed. Plus: yield mean = 1 to 1e-3 over
      1e5 draws; the `v` diversion; the cost formula; inputs consumed equal `a * y_tilde` capped at
      stock; `H = 1` when the enterprise's row of `a` is zero.

Downstream, not on this card's completion command but bound by it: **T-U1**, the per-period
conservation identity in `tests/unit/test_conservation.py`, is WO-009's must-pass test, and it can
only hold if the input-consumption line below is implemented exactly as written.

```bash
uv run pytest tests/unit/test_production.py -q
uv run ruff check gosplan/env/production.py
uv run ruff format --check gosplan/env/production.py
```

## Implementation notes

Two bodies. Both formulas are PLAN section 2.6 verbatim; **transcribe, do not re-derive.**

### 1. `coverage(X, need, weights, theta) -> H`

Takes `X` `(N, J)` input stocks on hand, `need` `(N, J)` this step's input need, `weights` `(N, J)`
with `omega_j = a_{s(i)j} / sum_j a_{s(i)j}`, and the complementarity exponent `theta`. Returns `H`
`(N,)` in `[0, 1]`.

```
H_ik = ( sum_j omega_j * min(1, X_ij / need_ikj)**(-theta) )**(-1/theta)
H_ik = 1                        if enterprise i requires no inputs
```

Edge cases, all of which must be handled **explicitly rather than by an epsilon**:

- `need_ikj == 0` contributes a coverage ratio of **1**. Not a division by zero, not a NaN, not a 0.
  A good that is not needed is fully covered by definition, and the same convention holds in the
  observation builder (WO-008).
- A whole row of `a` equal to zero gives `H = 1` for that enterprise.
- `theta = float("inf")` takes the `min` branch: `np.min` over the goods with **positive need only**.
  Do not compute the CES expression with a huge finite `theta` as a stand-in.
- A coverage ratio of exactly 0 with finite `theta` sends `0**(-theta)` to infinity and the whole
  aggregator to `H = 0`. That is the correct value; produce it **without emitting a numpy warning**,
  e.g. by handling the zero-ratio case before the power.
- `theta -> 1` reduces to the weighted harmonic mean of the clipped ratios; T-U7 checks it, so no
  special-casing is needed, only correctness at `theta = 1`.

### 2. `produce_step(state, action, cfg) -> (state, y_ik, c_ik)`

One PRODUCE step `k` for all `N` enterprises, in this order:

```
y_hat_ik   = (A_{s(i)} * cap_i / M) * e_ik           # M = cfg.incentive.steps_per_period
need_ikj   = a_{s(i)j} * y_hat_ik                    # for each good j with a_{s(i)j} > 0
H_ik       = coverage(X, need, omega, theta)
eps_ik     ~ LogNormal(-sigma_{s(i)}**2 / 2, sigma_{s(i)})        # mean exactly 1
             key = (seed_env, "yield", t, k, i)
y_tilde_ik = y_hat_ik * H_ik * eps_ik
y_ik       = y_tilde_ik * (1 - v_ik)                 # v = invest fraction; Phase 1 v == 0
X_ij      -= min(X_ij, a_{s(i)j} * y_tilde_ik)       # consumed against y_tilde, capped
c_ik       = kappa * e_ik**2 + F * 1[e_ik > 0] + kappa_q * q_ik * e_ik
```

Details that are easy to get wrong and are individually tested:

- **Inputs are consumed against `y_tilde`**, the output BEFORE the investment diversion, not against
  `y`. This is what makes the conservation identity T-U1 close.
- The consumption is **capped at the stock on hand**: `min(X_ij, a * y_tilde)`, so `X` never goes
  negative even when the coverage aggregator has already accounted for scarcity.
- The yield shock is drawn with `mean_log = -sigma**2 / 2` so that `E[eps] = 1`; the sigma is the
  **sector's**, `cfg.supply.yield_sigma[sector_of[i]]`, and the tuple is heteroskedastic by design.
  Draw one `(N,)` vector per step from the single key `(seed_env, "yield", t, k)` - the trailing
  enterprise index is what `shape` vectorises over.
- Phase-1 constants: `F = cfg.supply.setup_cost = 0.0` and `kappa_q = cfg.supply.quality_cost = 0.0`,
  so the cost is `kappa * e**2` in practice - but **write all three terms**, because the toggles are
  Phase-2 treatment variables.
- `cum_output` accumulates `y_ik` and `cum_cost` accumulates `c_ik`; both reset per period, and the
  reset belongs to the step machine (WO-009), not here.
- `pending_invest` receives `y_tilde_ik * v_ik` in Phase 2; in Phase 1 `v == 0` and the column is
  written with zeros rather than skipped.

### Phase-2 toggles that route through this function

All off in Phase 1. Implement the branch where PLAN section 2.6 states it and leave it inert; do not
invent behaviour beyond what is written there:

- setup cost `F`;
- increasing returns `A_j(Kap) = A_j * (Kap / Kap_0)**alpha_irs`;
- capital accumulation `Kap_{t+1} = (1 - dep) * Kap_t + matured investment`;
- I-O drift `a <- a * exp(zeta)`, `zeta ~ N(0, sigma_drift**2)`, with `State.planner_io` held FIXED
  so the planner's picture goes stale;
- quality routed through the bundle (`X_ij` credited as `deliv * qbar_j`);
- non-uniform `delivery_timing`, under which a delivered unit arrives at step
  `k ~ Categorical(arrival_probs)` drawn with purpose `arrival`.

Anything PLAN section 2.6 does not state about these branches is an AMBIGUITY REPORT, not a decision.

### Rule 7 is the standing constraint on this file

Nothing here may implement bunching, padding, storming, hoarding, shaving or trade directly.
**Storming in particular is a CONSEQUENCE** of `delivery_timing` plus the agent's effort choice,
never a rule that concentrates effort late in a period.

## Forbidden

- Any reference to reports, targets, claims, penalties, bonuses or rewards. This module computes
  physical output and cost and nothing else; if a formula here needs `T_i`, that is a defect in the
  formula, not a licence to import it.
- Any direct `numpy.random` or `jax.random` call. Every stochastic term goes through
  `gosplan.rng.draw` (CONTRACT rule 9), and no module-level generator exists anywhere.
- Reading `welfare_true`, `val_measured`, `consumer_delivery`, or any other enterprise's rows.
- Adding an epsilon to a denominator to dodge the `need == 0` case, or clipping `H` into a range the
  formula does not produce.
- Substituting a large finite `theta` for the `inf` branch.
- Importing scipy, pandas, jax or torch, at module scope or inside a function.
- Changing either signature, the module docstring, or the `Array` alias.
- Editing any test, `spec/spec.py`, `gosplan/config.py`, `gosplan/rng.py` or any other module.

## Contract rules that bite here

- **Rule 3 - STOP AND REPORT.** Choosing "the reasonable default" is a violation: do not pick an
  epsilon to avoid a division, do not clip a coverage ratio the spec does not clip, do not decide
  what a Phase-2 toggle does.
- **Rule 12 - WORK ORDERS (whitelist).** Read only the nine entries above, write only
  `gosplan/env/production.py`, run the completion command verbatim.
- **Rule 7 - NO HARD-CODED PATHOLOGY.** Storming is a consequence of delivery timing and the agent's
  effort choice; no rule here may concentrate effort, inflate a request or bank output. The lead
  reviews every `env/` diff against this rule before G0.
- **Rule 9 - RNG.** The yield shock goes through `gosplan.rng.draw` with purpose `yield`; no
  `numpy.random`, no module-level generator.
- **Rule 2 - FROZEN TESTS.** `tests/unit/test_production.py` is read-only; if it looks wrong, file an
  AMBIGUITY REPORT.
- **Rule 1 - FROZEN SPEC.** Signatures, the module docstring and the `Array` alias are frozen; this
  card fills bodies only.

## Definition of done

- [ ] `tests/unit/test_production.py` (T-U7 and the rest) green.
- [ ] `uv run ruff check . && uv run ruff format --check .` clean.
- [ ] `uv run python scripts/contract_guard.py --all` clean.
- [ ] No file outside "Write only" touched.
- [ ] CI `ci-ok` green on the PR.
- [ ] The four-part report posted as the PR body: 1) tests passed/failed with output; 2) ambiguity
      reports filed, each listed, or "none"; 3) files changed - expect exactly
      `gosplan/env/production.py`; 4) anything you were tempted to invent and did not - in particular
      any numerical guard around the coverage aggregator, and any Phase-2 branch whose behaviour PLAN
      section 2.6 does not pin.
